system-parachains/coretime/coretime-polkadot/src/migrations.rs

```diff
@@ -0,0 +1,330 @@
+			Decode::decode(&mut &state[..]).unwrap();
+
+		log::info!(target: TARGET, "Checking migration.");
+
+		let sale_info = SaleInfo::<Runtime>::get().unwrap();
+
+		// Check the sale start has not changed.
+		assert_eq!(sale_info, prev_sale_info);
+
+		// The workplan entries start from the region begin reported by the new SaleInfo.
+		let workplan_start = sale_info.region_begin;
+
+		// Check the reservations are still in the workplan out of an abundance of caution.
+		for (core_id, task) in [Task(1001), Task(1002), Task(1000), Task(1004), Task(1005)]
+			.into_iter()
+			.enumerate()
+		{
+			assert_eq!(
+				Workplan::<Runtime>::get((workplan_start, core_id as u16)),
+				Some(Schedule::truncate_from(Vec::from([ScheduleItem {
+					mask: CoreMask::complete(),
+					assignment: task,
+				}])))
+			);
+		}
+
+		// Make sure we've got all the leases.
+		let leases = Leases::<Runtime>::get();
+		assert_eq!(leases.len(), LEASES.iter().filter(|(_, l)| sale_info.region_end <= *l).count());
+
+		// Make something out of the leases that is easier to check against.
+		let leases_vec: Vec<(u32, u32)> =
+			leases.iter().map(|LeaseRecordItem { until, task }| (*task, *until)).collect();
+
+		// Iterate through hardcoded leases and check they're all correctly in state and scheduled
+		// in the workplan.
+		for (i, (para_id, until)) in LEASES.iter().enumerate() {
+			// Add the system parachains as an offset - these should come before the leases.
+			let core_id = i as u16 + 4;
+
+			assert!(leases_vec.contains(&(*until, *para_id)));
+
+			// This is the entry found in Workplan
+			let workload = Schedule::truncate_from(Vec::from([ScheduleItem {
+				mask: CoreMask::complete(),
+				assignment: Task(*para_id),
+			}]));
+
+			// They should all be in the workplan for next region.
+			assert_eq!(Workplan::<Runtime>::get((workplan_start, core_id)), Some(workload));
+		}
+
+		// For the leases we had before their lease should extend for an additional 11520
+		// timeslices (64 days).
+		for LeaseRecordItem { task, until } in prev_leases.iter() {
+			log::error!("{task}, {until}");
+			assert!(leases_vec.contains(&(*until + 11520, *task)))
+		}
+
+		// Iterate through hardcoded potential renewals and check they're all correctly in state
+		// and scheduled in the workplan.
+		for (i, (para_id, until)) in POTENTIAL_RENEWALS.iter().enumerate() {
+			// Add the system parachains as an offset - these should come before
+			// the leases.
+			let core_id = i as u16 + 4;
+
+			// This is the entry found in Workplan and PotentialRenewals.
+			let workload = Schedule::truncate_from(Vec::from([ScheduleItem {
+				mask: CoreMask::complete(),
+				assignment: Task(*para_id),
+			}]));
+
+			// Make sure they're not in the leases.
+			assert!(!leases.contains(&LeaseRecordItem { until: *until, task: *para_id }));
+
+			// Ensure they can renew in the next region.
+			assert_eq!(
+				PotentialRenewals::<Runtime>::get(PotentialRenewalId {
```

Comment:
We don't create a renewal for the entries in `POTENTIAL_RENEWALS` but here we assert they exist. I don't think we should create renewals on behalf of anyone. What I am missing?

Reply:
The migration was not finished at the point I handed it over. I hope these have now been added.